Shared/Supporting Files/Views/CategoryGridView.swift

```diff
@@ -38,8 +38,10 @@ struct CategoryGridView: View {
             LazyVGrid(columns: columns) {
                 ForEach(categories, id: \.self) { category in
                     NavigationLink {
-                        SampleListView(samples: samples.filter { $0.category == category })
-                            .navigationTitle(category)
+                        List {
```

Comment:
Why was `List` moved here? Now the name `SampleListView` is really confusing because it isn't actually a list. The list and navigation title really should be a part of that view and a better name for that view would be `CategoryView`, since it is a view that displays a category.

Reply:
I think Caleb moved it here to cope with the problem that the search result list need to  have different sections, while the category list doesn't need section. By changing "SampleListView" into a `ForEach`, it can deal with both of the cases. Do you have an alternative? This UI is seen in the Shortcuts and Mails app.

Another idea would be instead of putting the result lists into sections, they can be in separate lists switched by a segmented control. This UI is seen in Files and Music app.

Reply:
I think that the category view (with the list of all the samples in a given category) and the search results view should be two separate views. There's not a whole lot of overlap there. But as a single view, it seems like it could have a list but then consumers could change the list style as needed (`listStyle(_:)`).

Reply:
Thanks. I've made the following changes

- Remove `SampleListView` and moved the `SampleRow` view into a separate file
- Created `SearchResultView` to display the search results
- Inlined the sample list under a category grid item, and set its list style to "sidebar" (in CategoryGridView.swift)